src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1214,88 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
             return new Tuple<SqlConnectionInfo, DataTable, ServerConnection>(sqlConnInfo, dt, serverConnection);
         }
 
+        internal async Task HandleAgentNotebooksRequest(AgentNotebooksParams parameters, RequestContext<AgentNotebooksResult> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                try
+                {
+                    var result = new AgentNotebooksResult();
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                        parameters.OwnerUri,
+                        out connInfo);
+
+                    if (connInfo != null)
+                    {
+                        using (SqlConnection connection = new SqlConnection(ConnectionService.BuildConnectionString(connInfo.ConnectionDetails)))
+                        {
+                            connection.Open();
+                            /*
+                            This query fetches all the notebook Jobs across all the databases accessible by the user.
+                             */
+                            string getJobIdsFromDatabaseQueryString = 
+                                "DECLARE @script as varchar(MAX)\n" +
+                                "SET @script =\n" + 
+                                "'USE [?];\n" +
```

Comment:
>USE [?] [](start = 34, length = 7)

What if the user doesn't have permission to some DBs? You should probably do a try/catch so the entire thing doesn't fail in that case

Reply:
The system procedure "sp_MSforeachdb" runs it on only those dbs which user has permission to access. 
Code Referenced.
`EXEC sp_MSforeachdb @script`

Reply:
Where are you getting that information from? I looked around briefly and didn't see anything saying that specifically.

Regardless though - this procedure is specifically undocumented and unsupported so relying on behavior like that seems dangerous. There's no harm in being extra careful here I think.